src/EventStore.Client/StreamRevision.cs

```diff
@@ -23,6 +23,7 @@ namespace EventStore.Client {
 			_value = value;
 		}
 
+		public StreamRevision Next() => this + 1;
```

Comment:
Do you want to guard `ulong.MaxValue + 1` ? You could do something like this:

```
if (_value == ulong.MaxValue) then throw new InvalidOperationException("StreamRevision.None does not have a next value.");
return this + 1;
```

Reply:
It is checked here https://github.com/EventStore/EventStore-Client-Dotnet/pull/53/files/7fa5abac5aad4e2f1e54b16e78632f9d85b756d3..e9cc4e2d2764b9165b5951baa9ad6ec32ac2ef4f#diff-7c6948f8faf6bc76134dd921f85a525dR35